aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -110,6 +115,8 @@ private S3Client client() {
   @Override
   public void initialize(Map<String, String> properties) {
     this.awsProperties = new AwsProperties(properties);
+    this.writeTags = toTags(
```

Comment:
nit: I think we can combine this line into the helper method, that makes the purpose of the helper method clearer, like `this.writeTags = toTags(properties)`, and make the tag map extraction a part of the helper.